blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/Points.java

```diff
@@ -70,14 +70,9 @@ public int hashCode() {
         public boolean equals(Object obj) {
             if (obj == null || !(obj instanceof Point))
                 return false;
-            try {
-                Point<T> other = (Point<T>)obj;
-                return other.getTimestamp() == this.getTimestamp()
-                        && other.getData().equals(this.getData());
-            } catch (ClassCastException ex) {
-                // not a Point<T>, but a Point<X> instead?
-                return false;
-            }
+            Point<T> other = (Point<T>)obj;
+            return other.getTimestamp() == this.getTimestamp()
+                    && other.getData().equals(this.getData());
```

Comment:
Detail: If they pass, for example, a `Point<X>` to a `Point<Y>`, the cast won't through anything because both types get converted to just `Point`, due to type erasure of java generics. Then, in the `return` statement, `this.getData()` will return a `Y`, and pass it to `X.equals(Object)`. So the `equals` will receive the object as type `Object` and do whatever comparisons it wants to, and no cast exceptions will be thrown.